src/server/transport/socketio.ts

```diff
@@ -162,9 +169,33 @@ export class SocketIO {
             const { matchID } = this.clientInfo.get(socket.id);
             this.roomInfo.get(matchID).delete(socket.id);
             this.clientInfo.delete(socket.id);
+
+            if (!this.roomInfo.get(matchID).size) {
+              this.deleteMatchQueue(matchID);
+            }
           }
         });
       });
     }
   }
+
+  /**
+   * Create a PQueue for a given matchID.
+   * @param matchID
+   */
+  createMatchQueue(matchID: string): void {
+    if (!this.perMatchQueue.get(matchID)) {
+      this.perMatchQueue.set(matchID, new PQueue({ concurrency: 1 }));
+    }
+  }
+
+  /**
+   * Delete a PQueue for a given matchID.
+   * @param matchID
+   */
+  deleteMatchQueue(matchID: string): void {
+    if (this.perMatchQueue.get(matchID)) {
```

Comment:
As above, you can use `has` here.
```suggestion
    if (this.perMatchQueue.has(matchID)) {
```